docs/3.0. Behaviour.md

```diff
@@ -0,0 +1,45 @@
+# Behaviour
+
+## Introduction
+
+The purpose of this document is to explain how Sources, Flows and Senders within an NMOS compatible system can be reconfigured and EDIDs of Sinks associated with Receivers can be obtained.
+
+The terms ‘Device’, ‘Source’, ‘Flow’, ‘Sender’, ‘Receiver’ and ‘Sink’ will be used extensively throughout this document. Please refer to the core NMOS Technical Overview and Definition section for definitions of these.
+
+This API is intended to be used in conjunction with an NMOS IS-04 Discovery & Registration and NMOS IS-05 Connection Management deployment.
+
+## Definitions
+
+### Sink
+
+A media consuming unit associated with an IS-04 Receiver. Sink is a resource and based on Resource Core JSON Schema.
+
+### Media Profile
+
+Media description which describes a format acceptable for all Receivers implied to be a part of a connection. A Media Profile consist of separate parameters required to set up Sender and such IS-04 resources behind it as Source and Flow.
+
+## API Structure
+
+The API provides a mechanism to change the settings associated with logical Sources, Flows and Senders. Another use case is obtaining information about Sinks associated with Receivers. The UUIDs used to advertise Senders and Receivers in EDID Processing API must match those used in a corresponding IS-04 implementation.
+
+### Senders
+
+`/media-profiles` allows to pass an array of Media Profiles to a Sender and let a third-party which controls it choose what exact format to use. Also an API user can pass a single Media Profile to specify exact format strictly.
+
+The initial state of Media Profiles of any Sender is empty. Creating a connection with such Sender via IS-05 doesn't involve NMOS EDID Connection Management.
+
+GET request returns the last successfully applied Media Profiles array.
+
+PUT request MUST be validated and if the Device fully supports the proposed media parameters, it initiates an attempt to reconfigure Device and update according Flow and/or Source and/or Sender values. If success, the GET operation MUST start returning the accepted Media Profiles array, otherwise an error MUST be returned and nothing changed.
+
+DELETE request MUST clear the last successfully applied Media Profiles array and NMOS EDID Connection Management MUST fall out of use and return to its initial state.
+
+### Receivers
+
+`/sinks` lists IDs of Sinks associated with a specific Receiver.
+
+### Sinks
+
+`/sinks/{sinkId}` provides Sink resource for an API user. A Sink consists of Resource Core and EDID parsed into JSON.
```

Comment:
Slight modifications, the URLs looked like the same at first glance

```suggestion
### Receivers

`/receivers/{receiverId}/sinks` list of IDs corresponding to Sinks associated to the specific Receiver.

### Sinks

`/sinks/{sinkId}` provides the Sink resource which consists of Resource Core and EDID parsed into JSON.
```

Reply:
I like this change, but I have feeling that a verb is now missed in the 1st sentence.

Reply:
I would prefer this.

```suggestion
### Receivers

`/receivers/{receiverId}/sinks` returns a list of IDs corresponding to Sinks associated to the specific Receiver.

### Sinks

`/sinks/{sinkId}` provides the Sink resource which consists of Resource Core and EDID parsed into JSON.
```

Reply:
I like it!